### PR TITLE
Resource group supporting cgroups v2

### PR DESCRIPTION
**Pull request by SmartKeyerror**
After we changed the semantics of the resource group at #14562, it's time to add the implementation 
of Linux Cgroup v2.

In Linux Kernel, there is no difference between Cgroup v1 and Cgroup v2 in interior implementation,
the only change is the pseudo-files interface. Thanks to the #14343, we just need to add a new file
and implement a new `cgroupOpsRoutine`, which can be found at `get_group_routine_v2()`.

#### 1. Move all the resource group test files to the same directory, `isolation2/sql/resgroup`. 

In early times, we considered that developers will have different mount directories of Cgroup, so we
generate test files dynamically. But we found it is an over-design, so move some test files from `input/`
to `sql/`, and this will be convenient for developers.

#### 2. Implement Cgroup v2

This is been written at  `src/backend/utils/resgroup/cgroup-ops-linux-v2.c`. Cgroup version 2 uses 
the "unified hierarchy", all the presudo-file of different controllers are all in one directory, and not 
use the directyory "cpu, cpuset", "memory", etc.

This is the biggest difference between them, and directly decide how we configure them, here is an 
example:

```
# version 1
smart@stable:/sys/fs/cgroup$ ls
blkio  cpu  cpuacct  cpu,cpuacct  cpuset  devices  freezer  hugetlb  memory  net_cls  net_cls,net_prio  net_prio  perf_event  pids  rdma  systemd  unified

# version 2
[root@localhost gpdb]# ls
cgroup.controllers  cgroup.max.descendants  cgroup.type    cpu.stat         cpuset.cpus            io.max         pids.current
cgroup.events       cgroup.procs            cpu.idle       cpu.uclamp.max   cpuset.cpus.effective  io.pressure    pids.events
cgroup.freeze       cgroup.stat             cpu.max        cpu.uclamp.min   cpuset.cpus.partition  io.prio.class  pids.max
cgroup.kill         cgroup.subtree_control  cpu.max.burst  cpu.weight       cpuset.mems            io.stat        
cgroup.max.depth    cgroup.threads          cpu.pressure   cpu.weight.nice  cpuset.mems.effective  io.weight
```

The document of Cgroup v2 can be found at:

- [Control Group v2](https://docs.kernel.org/admin-guide/cgroup-v2.html)
- [Using cgroups-v2 to control distribution of CPU time for applications](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/managing_monitoring_and_updating_the_kernel/using-cgroups-v2-to-control-distribution-of-cpu-time-for-applications_managing-monitoring-and-updating-the-kernel)

#### 3. Rove the GUC of `gp_resource_group_enable_cgroup_version_two`

For now, if we want to enable Cgroup v2, we can use:

```
gpconfig -c gp_resource_manager -v group-v2
```

to config Greenplum database, and if we want use Cgroup v1, just use:

```
gpconfig -c gp_resource_manager -v group
```

### CODE REVIEW

**Comment by ashwinstar** [comment]
> For now, if we want to enable Cgroup v2, we can use:
> 
> gpconfig -c gp_resource_manager -v group-beta

What's the longer term plan to enable this. Would be good to put that in-place right now instead of delaying it. As once the feature becomes and available and field starts playing with it better to have the one which would be final one instead of temp. Will on longer term we will have it as `group-v2`?

**Comment by SmartKeyerror** [comment]
> What's the longer term plan to enable this.

I changed the value to `group-v2`.

**Comment by SmartKeyerror** [comment]
@50wu Hi, could you please take a look at this commit [add Cgroup v2 concourse pipeline](https://github.com/greenplum-db/gpdb/pull/14733/commits/ecff2533cc2e064d861da278087baf20acf5cb31) if you have time? Thanks.

**Review by 50wu — APPROVED** [review]
LGTM. Since you add a new job to the pipeline, please remember to sync up the pipeline configuration when merging the PR, also remember to update the gpdb_main_rhel8 and gpdb_main_oel8 pipelines. You can refer to the steps at https://github.com/greenplum-db/gpdb/tree/main/concourse/pipelines#create-and-update-production-pipelines

**Comment by kainwen** [comment]
Start reviewing.
Seems `src/backend/statistics/statistics_gram.c` mistakenly inclued, lets remove it from this commit.

**Comment by kainwen** [comment]
Why include this change `src/test/isolation2/sql/distributed_snapshot.sql`?

**Comment by SmartKeyerror** [comment]
> Why include this change `src/test/isolation2/sql/distributed_snapshot.sql`?

Already remove it.

File: src/backend/utils/resgroup/cgroup.c [hunk 1]
```diff
@@ -543,7 +583,7 @@ getCgroupMountDir()
 	{
 		char * p;
 
-		if (!gp_resource_group_enable_cgroup_version_two)
+		if (Gp_resource_manager_policy == RESOURCE_MANAGER_POLICY_GROUP)
```
Comment (kainwen) [inline src/backend/utils/resgroup/cgroup.c:586]: Should be `RESOURCE_MANAGER_POLICY_GROUP_V2` here?

Comment (SmartKeyerror) [reply src/backend/utils/resgroup/cgroup.c:586]: The original code is `!gp_resource_group_enable_cgroup_version_two`, means **not** Cgroup v2. After I added `RESOURCE_MANAGER_POLICY_GROUP_V2` to `ResourceManagerPolicy`, we just use `RESOURCE_MANAGER_POLICY_GROUP` to verify it's v1.

**Review by kainwen — COMMENTED** [review]
Go through all the code seems find.

Need to to through test case later.

**Review by kainwen — APPROVED** [review]

**Pull request merged**